File: backend/src/main/kotlin/com/plugga/backend/service/UserDeckServiceImpl.java
package com.plugga.backend.service;

import com.plugga.backend.dao.UserDeckDAO;
import com.plugga.backend.entity.UserDeck;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Repository;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Repository
public class UserDeckServiceImpl implements UserDeckService {
    private UserDeckDAO userDeckDAO;

    @Autowired
    public UserDeckServiceImpl(final UserDeckDAO userDeckDAO) {
        this.userDeckDAO = userDeckDAO;
    }

    @Override
    @Transactional
    public List<UserDeck> findAll() {
        return userDeckDAO.findAll();
    }

    @Override
    @Transactional
    public UserDeck findById(final int id) {
        return userDeckDAO.findById(id);
    }

    @Override
    @Transactional
    public List<UserDeck> findByDeckId(final int id) {
        return userDeckDAO.findByDeckId(id);
    }

    @Override
    @Transactional
    public List<UserDeck> findByUserId(final int id) {
        return userDeckDAO.findByUserId(id);
    }

    @Override
    @Transactional
    public void save(final UserDeck userDeck) {
        userDeckDAO.save(userDeck);
    }

    @Override
    @Transactional
    public void deleteById(final int id) {
        userDeckDAO.deleteById(id);
    }
}

File: backend/src/main/kotlin/com/plugga/backend/service/UserDeckServiceImpl.kt
package com.plugga.backend.service

import com.plugga.backend.entity.UserDeck
import com.plugga.backend.repository.UserDeckRepository
import java.util.Optional
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional

@Repository
class UserDeckServiceImpl @Autowired
constructor(private val userDeckRepository: UserDeckRepository) : UserDeckService {

    @Transactional
    override fun findAll(pageable: Pageable): Page<UserDeck> {
        return userDeckRepository.findAll(pageable)
    }

    @Transactional
    override fun findById(id: Int): UserDeck? {
        val queryResult: Optional<UserDeck> = userDeckRepository.findById(id)
        return if (queryResult.isPresent) queryResult.get() else null
    }

    @Transactional
    override fun findByDeckId(pageable: Pageable, id: Int): Page<UserDeck> {
        return userDeckRepository.findByDeckId(pageable, id)
    }

    @Transactional
    override fun findByUserId(pageable: Pageable, id: Int): Page<UserDeck> {
        return userDeckRepository.findByUserId(pageable, id)
    }

    @Transactional
    override fun save(userDeck: UserDeck) {
        userDeckRepository.save(userDeck)
    }

    @Transactional
    override fun deleteById(id: Int) {
        userDeckRepository.deleteById(id)
    }
}
